s2a/src/main/java/io/grpc/s2a/S2AChannelCredentials.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.s2a;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import io.grpc.Channel;
+import io.grpc.ChannelCredentials;
+import io.grpc.internal.ObjectPool;
+import io.grpc.internal.SharedResourcePool;
+import io.grpc.netty.InternalNettyChannelCredentials;
+import io.grpc.netty.InternalProtocolNegotiator;
+import io.grpc.s2a.channel.S2AHandshakerServiceChannel;
+import io.grpc.s2a.handshaker.S2AIdentity;
+import io.grpc.s2a.handshaker.S2AProtocolNegotiatorFactory;
+import java.util.Optional;
+import javax.annotation.concurrent.NotThreadSafe;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * Configures gRPC to use S2A for transport security when establishing a secure channel. Only for
+ * use on the client side of a gRPC connection.
+ */
+public final class S2AChannelCredentials {
+  /**
+   * Creates a channel credentials builder for establishing an S2A-secured connection.
+   *
+   * @param s2aAddress the address of the S2A server used to secure the connection.
+   * @return a {@code S2AChannelCredentials.Builder} instance.
+   */
+  public static Builder createBuilder(String s2aAddress) {
```

Comment:
S2AChannelCredentials is used to create gRPC ChannelCredentials that are configured to use S2A. s2aAddress is required because without it we would not be able to talk to S2A to get info needed to complete the handshake. We default to a plaintext connection between the client (application which creates ChannelCredentials using S2AChannelCredentials.createBuilder("s2a_addr")) and S2A because mTLS to S2A is not available in all environments. If the client wants and is able to use mTLS to S2A, they can set the s2aChannelCredentials. Please let me know if I missed something in addressing this comment.

An aside: probably will be getting rid of setS2AChannelCredentials as we will merge MtlsToS2AChannelCredentials and S2AChannelCredentials as you suggested in above comment.

cc: @matthewstevenson88